fastlane/Fastfile

```diff
@@ -766,79 +765,6 @@ platform :ios do
     )
   end
 
-  desc 'Download App Store Translations'
-  lane :download_app_store_translations do
-    files = {
-
-      # Metadata fields
-      'app_store_desc' => { desc: 'description.txt' },
-      'app_store_keywords' => { desc: 'keywords.txt' },
-      'app_store_subtitle' => { desc: 'subtitle.txt' },
-      'app_store_promo_text' => { desc: 'promotional_text.txt' },
-      'whats_new' => { desc: 'whats_new.txt' },
-
-      # Screenshots
-      'app_store_screenshot-1' => { desc: 'app_store_screenshot_1.txt' },
-      'app_store_screenshot-2' => { desc: 'app_store_screenshot_2.txt' },
-      'app_store_screenshot-3' => { desc: 'app_store_screenshot_3.txt' },
-      'app_store_screenshot-4' => { desc: 'app_store_screenshot_4.txt' },
-      'app_store_screenshot-5' => { desc: 'app_store_screenshot_5.txt' }
-    }
-
-    metadata_locales = GLOTPRESS_LOCALES
-
-    gp_downloadmetadata(project_url: 'https://translate.wordpress.com/projects/woocommerce%2Fwoocommerce-ios%2Frelease-notes/',
-                        target_files: files,
-                        locales: metadata_locales,
-                        source_locale: 'en-US',
-                        download_path: './fastlane/metadata')
-
-    # Ensure that localizations are present in the right places (some locale codes aren't correct)
-    {
-      'ar' => 'ar-SA',
-      'de' => 'de-DE',
-      'es' => 'es-ES',
-      'fr' => 'fr-FR',
-      'nl' => 'nl-NL',
-      'zh-cn' => 'zh-Hans',
-      'zh-tw' => 'zh-Hant'
-    }.each do |source, destination|
-      source_dir = File.join(Dir.pwd, 'metadata', source)
-      Dir.children(source_dir).each do |file|
-        source_file = File.join(source_dir, file)
-        destination_file = File.join(Dir.pwd, 'metadata', destination, file)
-        FileUtils.mv(source_file, destination_file)
-      end
-      FileUtils.remove_dir(source_dir)
-    end
-
-    # Copy source files into `en-US`
-    enUS_path = File.join(Dir.pwd, 'metadata', 'en-US')
-
-    FileUtils.mkdir_p(enUS_path)
-
-    [
-      'promo_screenshot_1.txt',
-      'promo_screenshot_2.txt',
-      'promo_screenshot_3.txt',
-      'promo_screenshot_4.txt',
-      'promo_screenshot_5.txt'
-    ].each do |filename|
-      source = File.join(Dir.pwd, 'appstoreres', 'metadata', 'source', filename)
-      destination = File.join(enUS_path, filename)
-      FileUtils.cp(source, destination)
-    end
-
-    # Replace the `default` directory with the contents of `en-US`
-    default_path = File.join(Dir.pwd, 'metadata', 'default')
-    FileUtils.remove_dir(default_path) if File.directory?(default_path)
-    FileUtils.copy_entry(enUS_path, default_path)
-
-    # Never store review information (it contains passwords and PII)
-    review_info_path = File.join(Dir.pwd, 'metadata', 'review_information')
-    FileUtils.remove_dir(review_info_path) if File.directory?(review_info_path)
-  end
-
```

Comment:
This lane has been here for a while, but as far as I can see it was unused. 

I say that because I couldn't find a call to it in the existing automation, even prior to my recent localization improvement changes (https://github.com/woocommerce/woocommerce-ios/commit/c991e2aadd23a7b6bfe6379cce8490213a096685). 

Up till then, the `finalize_release` lane called the release toolkit `ios_update_metadata` action. There also isn't any mention of the lane in the release scenario:
https://github.com/Automattic/mobile-toolbox/blob/f103711d97f2b26ef63cb5626c01daf712cce307/ios/wcios-release-scenario.sh#L89.

Reply:
This lane is supposed to be used when we need to create promo screenshots though, to ensure that the copies used during `create_promo_screenshots` below will be up-to-date.

Maybe (probably) we should make `create_promo_screenshots` explicitly call that `download_app_store_translations ` lane as the first thing it does before actually calling `promo_screenshots`. The drawback of doing this is that, when we update the promo screenshots, usually we need to do some trial-and-error iterations to get the `screenshots.json` file right and be happy with the rendered images produced by it, and it would be a bit painful to have to re-download the strings file every time if we call `create_promo_screenshots` every 5 minutes we try to iterate on it. I guess that might be why we didn't make `create_promo_screenshots` call that `download_app_store_translations` lane in the first place.

That being said, I'd suggest to have `create_promo_screenshots` call `download_app_store_translations` by default, but accept an option like `skip_translations_download` to bypass it while developers iterate to adjust the new promo screenshots when we modify them (on other words, have `download_app_store_translations unless options[:skip_translations_download]` in `create_promo_screenshots`.

---

Cc @rachelmcr as she's actually working on promo screenshots for WCiOS in https://github.com/woocommerce/woocommerce-ios/pull/6933 (and would definitively not want that `download_app_store_translations` lane to be deleted!)

Reply:
@mokagio Also, now that I saw [your _other_ PR #6937](https://github.com/woocommerce/woocommerce-ios/pull/6937/files), which adds the `app_store_screenshot_*.txt` files to the list in the other `download_localized_metadata_from_glotpress` lane, then **_maybe_** we could get rid of this `download_app_store_translations` lane (which does roughtly the same thing as `download_localized_metadata_from_glotpress`) and call `download_localized_metadata_from_glotpress` in `create_promo_screenshots` instead.

That being said, I think the `download_localized_metadata_from_glotpress` lane misses the tweaks we have in this `download_app_store_translations` which ensures that some source-of-truth files from `appstoreres/metadata/source` are copied over to `fastlane/metadata/en-US`.

The ideal solution for all this mess imho would be to move the source of truth for the screenshots metadata from `fastlane/appstoreres/metadata/source/*.txt` to `fastlane/metadata/en-US` for consistency, and update all the relevant screenshot-related lanes to match that change. That sould make the whole shenanigans around those more consistent and easier to understand, instead of having 2 different source of truths for the original copies of the screenshots, and 2 different lanes doing partially the same thing.
But we must be careful when we'll do that change to not break anything screenshot-related and make sure that every tweak for those screenshot `*.txt` files made in `download_app_store_translations` would either be confirmed to not be necessary anymore, or be ported into the `download_localized_metadata_from_glotpress` lane if they still need special treatment (because of them being used by the screenshots composer and not by `deliever` like the rest of them)

\cc @rachelmcr

Reply:
> This lane is supposed to be used when we need to create promo screenshots though

Ah, I didn't know that... I guess I could have also MGSed as part of my research 🤔 If I'd done that, I would have seen @rachelmcr recent mentions of this in P2 and some old chatter in Slack.

I think it's best to revert this removal for now. I'm hesitant to say "I'll look into rewiring this soon" because we have release finalizations and code freezes coming up next, plus this is but a cleanup. Still, agreed we ought to tidy this up sooner rather than later.